core/trace/dt_span_serializer_mapper.go

```diff
@@ -179,7 +180,7 @@ func getProtoLinks(links []sdktrace.Link, tenantId, clusterId int32) ([]*protoTr
 		}
 
 		if spanContext.IsRemote() {
-			if fw4Tag, err := fw4.GetMatchingFw4FromTracestate(spanContext.TraceState(), tenantId, clusterId); err != nil {
+			if fw4Tag, err := fw4.GetMatchingFw4FromTracestate(spanContext.TraceState(), qualifiedTenantId.TenantId, qualifiedTenantId.ClusterId); err != nil {
```

Comment:
I would change the parameter type of GetMatchingFw4FromTracestate. Otherwise you still have the same chance of accidentally swapping the arguments.